src/main/java/org/apache/mesos/specification/TaskTypeSpecification.java

```diff
@@ -1,8 +1,16 @@
 package org.apache.mesos.specification;
 
+import org.apache.mesos.Protos;
+
+import java.util.Collection;
+
 /**
- * A TaskTypeSpecification describes the count of a particular TaskSpecification which should be launched.
+ * A TaskTypeSpecification describes a Type of Task and how many should be launched.
  */
-public interface TaskTypeSpecification extends TaskSpecification {
+public interface TaskTypeSpecification {
```

Comment:
Name is [taken care of here](https://github.com/mesosphere/dcos-commons/blob/master/src/main/java/org/apache/mesos/specification/DefaultPhaseSpecification.java#L33) or through other user implementations, although I don't know why any service would ever want to be opinionated on the matter of `Task` name.

I don't see any reason `Resources` would need ID specifiable resources.  It's sort of the defining characteristic of a `TaskType` that all instances of it consume the same `Resources`.  For Volumes it's possible that a per `Task` instance volume container-path might be desirable, but since each is running in their own container (Executor) I don't see it as a necessity.  So far of the services we've encountered the only necessary change is on a Command basis, and none of the services have any special preference regarding `Task` name.

Make sense?